.github/workflows/publishing.yml

```diff
@@ -0,0 +1,53 @@
+name: publishing
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - '*'
+  workflow_dispatch:
+
+env:
+  ROCK_NAME: vshard
+
+jobs:
+  rockspec-scm:
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Push rockspec
+        uses: tarantool/rocks.tarantool.org/github-action@master
+        with:
+          auth: ${{ secrets.ROCKS_AUTH }}
+          files: ${{ env.ROCK_NAME }}-scm-1.rockspec
+
+  version-check:
+    # We need to run this job only on tag push.
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check module version
+        uses: tarantool/actions/check-module-version@master
+        with:
+          module-name: vshard
+
+  rockspec-tag:
+    if: startsWith(github.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    needs: version-check
+    steps:
+      - uses: actions/checkout@v3
+      - name: Patch scm rockspec
+        run: |
+          sed \
+            -e "s/branch = '.\+'/tag = '${{ github.ref_name }}'/g" \
+            -e "s/version = '.\+'/version = '${{ github.ref_name }}-1'/g" \
+            ${{ env.ROCK_NAME }}-scm-1.rockspec > \
+            ${{ env.ROCK_NAME }}-${{ github.ref_name }}-1.rockspec
+      - name: Push rockspec
+        uses: tarantool/rocks.tarantool.org/github-action@master
+        with:
+          auth: ${{ secrets.ROCKS_AUTH }}
+          files: ${{ env.ROCK_NAME }}-${{ github.ref_name }}-1.rockspec
```

Comment:
I would also publish `.all.rock` to speed up the installation on the user side. See https://github.com/tarantool/expirationd/blob/bcbe8c46ceca8d82dc5bb66b1d0a40deca03ad9d/.github/workflows/publish.yml#L43-L71 for example.

Reply:
Makes sense, thank you!

Reply:
Fixed.